eclair-core/src/test/scala/fr/acinq/eclair/integration/basic/zeroconf/ZeroConfActivationSpec.scala

```diff
@@ -60,8 +60,13 @@ class ZeroConfActivationSpec extends FixtureSpec with IntegrationPatience {
     assert(!bob.nodeParams.features.activated.contains(ZeroConf))
 
     val channelId = createChannel(f)
-    assert(!getChannelData(alice, channelId).asInstanceOf[ChannelDataWithCommitments].commitments.params.channelFeatures.hasFeature(ZeroConf))
-    assert(!getChannelData(bob, channelId).asInstanceOf[ChannelDataWithCommitments].commitments.params.channelFeatures.hasFeature(ZeroConf))
+    Seq(alice, bob).foreach { node =>
+      getChannelData(node, channelId) match {
+        case d: DATA_WAIT_FOR_DUAL_FUNDING_SIGNED => assert(!d.channelParams.channelFeatures.hasFeature(ZeroConf))
+        case d: ChannelDataWithCommitments => assert(!d.commitments.params.channelFeatures.hasFeature(ZeroConf))
+        case d => fail(s"unexpected channel state: ${d.getClass.getSimpleName}")
+      }
+    }
```

Comment:
Since this seems to be flaky due to a race condition, wouldn't using `eventually` like is done for later tests that use `createChannel` make more sense? For example, change `ChannelDataWithCommitments`  to `ChannelDataWithoutCommitments`:

```scala
    eventually {
      assert(!getChannelData(alice, channelId).asInstanceOf[ChannelDataWithoutCommitments].channelParams.channelFeatures.hasFeature(ZeroConf))
      assert(!getChannelData(bob, channelId).asInstanceOf[ChannelDataWithCommitments].commitments.params.channelFeatures.hasFeature(ZeroConf))
    }

```

It's equivalent to your change, but I found it easier to understand what's going on relative to the other tests.

Reply:
Oh! right, you can actually use exactly what the other tests use. That's even clearer than what I suggested.